app/services/UserService.scala

```diff
@@ -232,12 +232,47 @@ class UserService @Inject() (
   }
 
   def inactiveActiveFlag(profile: Profile) = {
-    profileDao.setActiveFlag(profile, Some("inactive")).map(_ match{
+    profileDao.setActiveFlag(profile, Some("inactive")).map(_ match {
+      case true => "inactive"
+      case false => None
+    })
+  }
+
+  /** checkActiveFlag
+    * check if the active flag is set, not set or requested
+    */
+  def checkActiveFlag(profile: Profile) = {
+    //the conditions. Actually we support only one Crew
+    profileDao.getProfile(profile.email.head).flatMap({
+      case Some(p) => {
+        val primaryCrew = hasPrimaryCrew(p)
+        val status = p.supporter.active match {
+          case Some(status) => status
+          case _ => "inactive"
+        }
+        Future.successful(Json.obj("status" -> status, "conditions" ->
+          Json.obj(
+            "hasCrew" -> primaryCrew
+          )
+        ))
+      }
+      case _ => Future.successful(Json.obj("status" -> "inactive", "conditions" -> Json.obj("hasCrew" -> false)))
+    })
+  }
+
+  def activeNVM(profile: Profile) = {
+    profileDao.setNVM(profile, Some(System.currentTimeMillis() + 1000*60*60*365)).map(_ match {
+      case true => "active"
```

Comment:
Same here as before:
```suggestion
      case true => Some("active")
```